va/validation-authority_test.go

```diff
@@ -258,23 +232,33 @@ func TestSimpleHttpTLS(t *testing.T) {
 }
 
 func TestSimpleHttp(t *testing.T) {
-	va := NewValidationAuthorityImpl(true)
+	va := NewValidationAuthorityImpl(false)
 	va.DNSResolver = &mocks.MockDNS{}
 
 	tls := false
 	chall := core.Challenge{Type: core.ChallengeTypeSimpleHTTP, Token: expectedToken, TLS: &tls}
 
+	// NOTE: We do not attempt to shut down the server. The problem is that the
+	// "wait-long" handler sleeps for ten seconds, but this test finishes in less
+	// than that. So if we try to call hs.Close() at the end of the test, we'll be
+	// closing the test server while a request is still pending. Unfortunately,
+	// there appears to be an issue in httptest that trips Go's race detector when
+	// that happens, failing the test. So instead, we live with leaving the server
+	// around till the process exits.
+	hs := simpleSrv(t, expectedToken, tls)
+
+	port, err := getPort(hs)
+	test.AssertNotError(t, err, "failed to get test server port")
+
+	// Attempt to fail a challenge by telling the VA to connect to a port we are
+	// not listening on.
+	va.simpleHTTPPort = (port + 1) % 65536
```

Comment:
How about trying to grab a port with net.Listen("tcp", "localhost:0") but without putting an HTTP server on it?

I'm a little worried about doing this without trying to grab the port for ourselves because we could end up Doing Things to services running on their machines that maybe can't handle having HTTP splatted at them.

I'm thinking about those times we'd cause Cassandra to eat an entire heap by nc'ing "hello" to it. I'm imagining devs have all sorts of weird crap on their systems.